### PR TITLE
fix composite.merge for indices

### PR DESCRIPTION
**Pull request by kenshyx**
## Description

The merge method from Composite class discards all the additional indices from the merged sources.
This patch fixes it by ensuring that the indices are being passed to the nextDefinition

## How Has This Been Tested?

I've used this package versions with my local inmemory configuration
```
    "@ceramicnetwork/cli": "2.36.0-rc.2",
    "@ceramicnetwork/common": "2.31.0-rc.1",
    "@ceramicnetwork/http-client": "2.28.0-rc.1",
    "@composedb/cli": "0.5.0-rc.2",
    "@composedb/client": "0.5.0-rc.2",
    "@composedb/devtools": "0.5.0-rc.2",
    "@composedb/devtools-node": "0.5.0-rc.2"
```

## Definition of Done

Before submitting this PR, please make sure:

- [x] The work addresses the description and outcomes in the issue
- [x] I have added relevant tests for new or updated functionality
- [x] My code follows conventions, is well commented, and easy to understand
- [x] My code builds and tests pass without any errors or warnings
- [x] I have tagged the relevant reviewers
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation
- [x] The changes have been communicated to interested parties

### CODE REVIEW

**Comment by kenshyx** [comment]
@PaulLeCam

**Review by PaulLeCam — COMMENTED** [review]
Thanks!
Could you add or update existing tests as well to ensure the behavior is correctly applied please?

LGTM but I'll let @dbcfd comment on the logic, it might need something more complex than and shallow merge to ensure all the indices are present.

**Review by dbcfd — APPROVED** [review]

**Comment by kenshyx** [comment]
This can be merged only by someone with write access to the repo.

**Comment by PaulLeCam** [comment]
Merged, thanks!

**Comment by PaulLeCam** [comment]
@kenshyx it's now published in RC3

**Comment by kenshyx** [comment]
Nice, thanks! :raised_hands:

**Pull request merged**
